span_test.go

```diff
@@ -395,6 +394,16 @@ func TestSpanContextRaces(t *testing.T) {
 	go accessor(func() {
 		span.BaggageItem("k")
 	})
+	go accessor(func() {
+		ext.SamplingPriority.Set(span, 0)
+	})
+	go accessor(func() {
+		EnableFirehose(span)
+	})
+	go accessor(func() {
+		span.SpanContext().samplingState.setFlag(flagDebug)
+	})
```

Comment:
I believe `reset` still races, but unsure if this is worth protecting.  This is only called when the syncpool is enabled and generally the lifecycle code seems dangerous.  It would be trivial for the calling application to hold a `*Span`  after `.Finish()` and do terrible things with it so I think we have to trust the calling code.